From: Alexander Larsson <alexl@redhat.com>
Date: Mon, 16 Dec 2019 20:43:59 +0100
Subject: dir: Fix leak

When recreating self->repo, also clear self->cache_dir, otherwise it was
leaking when we replaced it.

Origin: https://github.com/flatpak/flatpak/pull/3307
---
 common/flatpak-dir.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/common/flatpak-dir.c b/common/flatpak-dir.c
index 0fd125d..9594f17 100644
--- a/common/flatpak-dir.c
+++ b/common/flatpak-dir.c
@@ -2804,6 +2804,9 @@ flatpak_dir_recreate_repo (FlatpakDir   *self,
   gboolean res;
   OstreeRepo *old_repo = g_steal_pointer (&self->repo);
 
+  /* This is also set by ensure repo, so clear it too */
+  g_clear_object (&self->cache_dir);
+
   res = flatpak_dir_ensure_repo (self, cancellable, error);
   g_clear_object (&old_repo);
 
